0027-managed-range-cache-range-lru.md

```diff
@@ -0,0 +1,58 @@
+---
+feature: managed_range_cache_range_lru
+authors:
+  - "jon-chuang"
+start_date: "2022/12/01"
+---
+
+# The Design of `ManagedRangeCache`'s Last-access-time Oracle - `RangeLru`
+
+## Summary
+The design of a mechanism for evicting ranges last accessed before a certain epoch for the `DynamicFilter` operator
+
+## Details
+
+The dynamic filter requests a given range at every epoch. As this range is the difference between the previous and current pointer, the path of these ranges as a function of the epoch is always continuous.
+
+The current `DynamicFilter` cache, the `RangeCache` differs from other caches in that it caches logical ranges rather than physical elements. That's because the basic unit of storage access for `DynamicFilter` are logical ranges rather than concrete keys.
+
+At any point in time, the `RangeCache` owns a contiguous range.
+
+In-line with our non-OOM and memory-efficiency requirements, we require a way to evict ranges that have not been requested by the operator for a long time. We choose to do so in a similar fashion to `ManagedLru` - based on the last-accessed time of a given subrange (the subranges being a partition of the total range owned by `RangeCache`). 
```

Comment:
Btw, why not maintain the epoch per row?

Reply:
> Btw, why not maintain the epoch per row?

This would violate the integrity of range cache since it guarantees to cache all rows in a certain range.

Reply:
> Btw, why not maintain the epoch per row?

Actually, on further thought, perhaps ManagedLru cache can work since it will update all the rows in a range with new timestamp on every read. So it will achieve the same thing but be less efficient since it has to track each row rather than the access of logical range.